README.md

```diff
@@ -54,36 +73,62 @@ if __name__ == "__main__":
     main(MyCharm)
 ```
 
-This charm does nothing, though, so you'll typically want to observe some Juju
-events, such as `start`:
+This charm does nothing, because the `MyCharm` class passed to the operator
+framework's `main` function is empty. Functionality can be added to the charm
+by instructing it to observe particular Juju events when the `MyCharm` object
+is initialized. In the example below we're observing the `start` event. The
+name of the event is a function of `self.on`, so in this case `self.on.start`
+is passed to `self.framework.observe`. This will look for method named
+`on_start` when that Juju event is triggered.
 
 ```python
 class MyCharm(CharmBase):
     def __init__(self, *args):
         super().__init__(*args)
-        self.framework.observe(self.on.start, self.on_start)
+        self.framework.observe(self.on.start, self)
 
      def on_start(self, event):
-        # Handle the event here.
+        # Handle the start event here.
 ```
 
 Every standard event in Juju may be observed that way, and you can also easily
-define your own events in your custom types.
+define your own events in your custom types. For example, to observe the
+`config-changed` Juju event:
 
-The `hooks/` directory will then contain symlinks to your `src/charm.py` entry
+```python
+class MyCharm(CharmBase):
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.framework.observe(self.on.start, self)
+        self.framework.observe(self.on.config_changed, self)
+
+     def on_start(self, event):
+        # Handle the start event here.
+
+     def on_config_changed(self, event):
+        # Handle the config-changed event here.
+```
+
+The `hooks/` directory must contain a symlink to your `src/charm.py` entry
 point so that Juju can call it. You only need to set up the `hooks/install` link
 (`hooks/start` for K8s charms, until [lp#1854635](https://bugs.launchpad.net/juju/+bug/1854635)
```

Comment:
I think it is a good opportunity to mention that storage events for [non-kubernetes charms](https://discourse.jujucharms.com/t/install-hook-now-called-for-kubernetes-charms/2585) run before the `install` event so symlinks for those need to be created as well.

> If your application depends on Juju storage availability before it can process the install hook, you also need to create `<storage-name>-storage-attached` symlinks as they run before the `install` event on non-Kubernetes models.

Reply:
Let's please not hold @mthaddon's collaboration based on what _else_ might be improved on the README. Those can come in a separate PR.